draft-packages/pagination/KaizenDraft/Pagination.tsx

```diff
@@ -0,0 +1,168 @@
+
+  const truncateIndicator = (page: number) => <TruncateIndicator key={page} />
+
+  const pagination = () => {
+    const items: JSX.Element[] = []
+
+    // truncateSize is 1 now but could be 0 if we add the ability to hide it.
+    const truncateSize = 1
+
+    // Simplify generation of pages if number of available items is equal or greater than total pages to show
+    if (
+      1 + 2 * truncateSize + 2 * siblingPagesRange + 2 * boundaryPagesRange >=
+      pageCount
+    ) {
+      const allPages = createRange(1, pageCount).map(paginationIndicator)
+      items.push(...allPages)
+    } else {
+      // Calculate group of first pages
+      const firstPagesStart = 1
+      const firstPagesEnd = boundaryPagesRange
+      const firstPages = createRange(firstPagesStart, firstPagesEnd).map(
+        paginationIndicator
+      )
+      // Calculate group of last pages
+      const lastPagesStart = pageCount + 1 - boundaryPagesRange
+      const lastPagesEnd = pageCount
+      const lastPages = createRange(lastPagesStart, lastPagesEnd).map(
+        paginationIndicator
+      )
+      // Calculate group of main pages
+      const mainPagesStart = Math.min(
+        Math.max(
+          currentPage - siblingPagesRange,
+          firstPagesEnd + truncateSize + 1
+        ),
+        lastPagesStart - truncateSize - 2 * siblingPagesRange - 1
+      )
+      const mainPagesEnd = mainPagesStart + 2 * siblingPagesRange
+      const mainPages = createRange(mainPagesStart, mainPagesEnd).map(
+        paginationIndicator
+      )
+
+      // Add group of first pages
+      items.push(...firstPages)
+
+      // Calculate and add truncate before group of main pages
+      const firstEllipsisPageNumber = mainPagesStart - 1
+      const showPageInsteadOfFirstEllipsis =
+        firstEllipsisPageNumber === firstPagesEnd + 1
+      items.push(
+        showPageInsteadOfFirstEllipsis
+          ? paginationIndicator(firstEllipsisPageNumber)
+          : truncateIndicator(firstEllipsisPageNumber)
+      )
+
+      // Add group of main pages
+      items.push(...mainPages)
+
+      // Calculate and add truncate after group of main pages
+      const secondEllipsisPageNumber = mainPagesEnd + 1
+      const showPageInsteadOfSecondEllipsis =
+        secondEllipsisPageNumber === lastPagesStart - 1
+      items.push(
+        showPageInsteadOfSecondEllipsis
+          ? paginationIndicator(secondEllipsisPageNumber)
+          : truncateIndicator(secondEllipsisPageNumber)
+      )
+
+      // Add group of last pages
+      items.push(...lastPages)
+    }
+    return items
+  }
+
+  const previousPageDisabled = currentPage <= 1
+  const nextPageDisabled = currentPage >= pageCount
+
+  return (
+    <div className={styles.container}>
+      <button
+        className={cx(styles.arrowIconWrapper, {
+          [styles.arrowIconWrapperDisabled]: previousPageDisabled,
+        })}
+        aria-label={ariaLabelPreviousPage}
+        aria-disabled={previousPageDisabled}
```

Comment:
I'm pretty sure `disabled` should be used in favor of `aria-disabled` as `<button>` natively supports the former.
Disagree with me though, i'm not 100% sure.